cpp/src/arrow/dataset/parquet_encryption_config.h

```diff
@@ -29,44 +29,50 @@ struct DecryptionConfiguration;
 namespace arrow {
 namespace dataset {
 
-/// \brief Core configuration class encapsulating parameters for high-level encryption
-/// within Parquet framework.
-///
-/// ParquetEncryptionConfig serves as a bridge, passing encryption-related
-/// parameters to appropriate components within the Parquet library. It holds references
-/// to objects defining encryption strategy, Key Management Service (KMS) configuration,
-/// and specific encryption configurations for Parquet data.
-///
-/// \member crypto_factory Shared pointer to CryptoFactory object, responsible for
-/// creating cryptographic components like encryptors and decryptors. \member
-/// kms_connection_config Shared pointer to KmsConnectionConfig object, holding
-/// configuration parameters for connecting to a Key Management Service (KMS).
-/// \member encryption_config Shared pointer to EncryptionConfiguration object, defining
-/// specific encryption settings for Parquet data, like keys for different columns.
 struct ARROW_DS_EXPORT ParquetEncryptionConfig {
   std::shared_ptr<parquet::encryption::CryptoFactory> crypto_factory;
   std::shared_ptr<parquet::encryption::KmsConnectionConfig> kms_connection_config;
   std::shared_ptr<parquet::encryption::EncryptionConfiguration> encryption_config;
+  /// \brief Core configuration class encapsulating parameters for high-level encryption
```

Comment:
I think we typically pt those docs just above the struct? (as you had initially)

Reply:
@jorisvandenbossche I moved them below bc `/property` was not able to find the properties. But, perhaps we need to be using a different doxygen keyword. Which do you recommend?

(Or maybe I was wrong, and they weren't finding them bc the property name needed to be on a different line.)

Reply:
Here is the error:

> /arrow/cpp/src/arrow/dataset/parquet_encryption_config.h:40: error: documented symbol 'arrow::dataset::crypto_factory' was not declared or defined. (warning treated as error, aborting now)

So perhaps `/property` is not the appropriate keyword. (@tolleybot `/member` did not seem to exist.)

Reply:
@jorisvandenbossche The examples I see here: https://github.com/apache/arrow/blob/c3fe65a8bd700231d9d40aa9638304706e8f0d99/cpp/src/arrow/dataset/file_parquet.h#L275 throughout this file are to put the general descriptor of the struct above the struct, and then the descrptions of the properties within the struct, above the property. 

I am going to follow that approach, let me know if you would like something different!

Reply:
I actually have a commit, and ran into a protected branch error when trying to push it. I am going to open a PR!

Reply:
https://github.com/tolleybot/arrow/pull/3